src/websocket/webSocketClient.ts

```diff
@@ -0,0 +1,131 @@
+import { Api } from "coder/site/src/api/api";
```

Comment:
What do you think about merging this with the rest client? That would match how all the web socket and regular API request methods are together in `coder/coder`.  Then we can pass around just the one rest client and do `restClient.watchWorkspace()`, etc (we would have to extend `Api` of course).